wagtail_localize/modeladmin/views.py

```diff
@@ -0,0 +1,164 @@
+from django.contrib.auth.decorators import login_required
+from django.core.exceptions import ImproperlyConfigured
+from django.shortcuts import get_object_or_404
+from django.utils.decorators import method_decorator
+from wagtail import VERSION as WAGTAIL_VERSION
+from wagtail.contrib.modeladmin.views import (
+    ChooseParentView,
+    CreateView,
+    DeleteView,
+    EditView,
+    IndexView,
+    InspectView,
+)
+from wagtail.core.models import Locale, TranslatableMixin
+from wagtail.utils.version import get_main_version
+
+from wagtail_localize.models import Translation
+from wagtail_localize.views import edit_translation
+
+
+if WAGTAIL_VERSION >= (2, 15):
+    from wagtail.contrib.modeladmin.views import HistoryView
+else:
+    HistoryView = object
+
+
+class TranslatableViewMixin:
+    def __init__(self, *args, **kwargs):
+        self.locale = None
+        super().__init__(*args, **kwargs)
+
+        if not issubclass(self.model, TranslatableMixin):
+            raise ImproperlyConfigured(
+                f"Model `{self.model}` used in translatable view `{self.__class__}` "
+                f"must subclass the `{TranslatableMixin}` class."
+            )
+
+    @method_decorator(login_required)
+    def dispatch(self, request, *args, **kwargs):
+        if getattr(self, "instance", None):
+            self.locale = self.instance.locale
+        if "locale" in request.GET:
+            self.locale = get_object_or_404(Locale, language_code=request.GET["locale"])
+        if not self.locale:
+            self.locale = Locale.get_active()
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["locale"] = self.locale
+        context["translations"] = []
+        context["wagtail_version"] = get_main_version()
+        return context
+
+
+class TranslatableIndexView(TranslatableViewMixin, IndexView):
+    def get_filters(self, request):
+        filters = super().get_filters(request)
+        filters[2]["locale"] = self.locale.id
```

Comment:
nitpick: this warrants at least a comment as to why `filters[2]`. Had to double check myself. `get_filters()` returns `(filter_specs, bool(filter_specs), lookup_params, use_distinct)`

Reply:
Added the extra comment.